Features/Intel/UserInterface/VirtualKeyboardFeaturePkg/VirtualKeyboardDxe/Keyboard.c

```diff
@@ -897,9 +897,44 @@ VkApiStart (
     goto Error;
   }
 
+  for (KeyData.Key.ScanCode = SCAN_UP; KeyData.Key.ScanCode <= SCAN_F12; KeyData.Key.ScanCode++) {

```

Comment:
This loop can be extended to include `SCAN_ESC` and remove the call to `RegisterKeyNotify()` above for `SCAN_ESC`.

Reply:
please double check I updated the bounds correctly, I could not find the scan code definitions